src/provinces/province.cpp

```diff
@@ -1567,7 +1573,7 @@ std::vector<dcon::province_id> make_land_path(sys::state& state, dcon::province_
 			if((bits & province::border::impassible_bit) == 0 && !origins_vector.get(other_prov)) {
 				if(other_prov == end) {
 					fill_path_result(nearest.province);
-					return path_result;
+					return assert_path_result(path_result);
```

Comment:
I believe that this will break NRVO

Reply:
it shalln't in release, in debug it may, but that's the point - to assert all paths are being checked correctly

Reply:
It will break in release too. NRVO requires the return statement to use the *name* (NRVO = named return value optimization). This is a case where you would need to use a macro. In fact, a macro with the loop wrapped in NDEBUG just in case